sphinx/writers/latex.py

```diff
@@ -330,16 +330,20 @@ def get_colspec(self):
         elif self.colwidths and 'colwidths-given' in self.classes:
             total = sum(self.colwidths)
             colspecs = ['\\X{%d}{%d}' % (width, total) for width in self.colwidths]
-            return '{|%s|}\n' % '|'.join(colspecs)
+            return '{%s%s%s}\n' % (self.colsep, self.colsep.join(colspecs),
+                    self.colsep)
+
         elif self.has_problematic:
-            return '{|*{%d}{\\X{1}{%d}|}}\n' % (self.colcount, self.colcount)
+            return '{%s*{%d}{\\X{1}{%d}%s}}\n' % (self.colsep, self.colcount,
+                    self.colcount, self.colsep)
         elif self.get_table_type() == 'tabulary':
             # sphinx.sty sets T to be J by default.
-            return '{|' + ('T|' * self.colcount) + '}\n'
+            return '{' + self.colsep + (('T' + self.colsep) * self.colcount) + '}\n'
         elif self.has_oldproblematic:
-            return '{|*{%d}{\\X{1}{%d}|}}\n' % (self.colcount, self.colcount)
+            return '{%s*{%d}{\\X{1}{%d}%s}}\n' % (self.colsep, self.colcount,
```

Comment:
The definitions of the `\X` and `\Y` column types in sphinxlatextables.sty need adjustment probably as they use `\arrayrulewidth` (except if booktabs set this to zero, I must check; but about this I think you must allow not using vertical rules without tying user to using booktabs package, so the better terminology throughout the LaTeX writer would be not to refer to booktabs but to something such as "nocolsep").  Some thinking required, depositing this comment here.